third_party/amd/python/triton_amd.cc

```diff
@@ -121,6 +126,66 @@ void addControlConstant(llvm::Module *module, const char *name,
 
 LLD_HAS_DRIVER(elf)
 
+static void checkMatmulConstraints(const std::string &A_dtype,
```

Comment:
We should unify and reuse the copy from `triton_nvidia.cc` to avoid duplication.

Reply:
mi350 fp8 variants are OCP so I'm surprised that they don't support.. There might be some inconsistency. You can ask internally.